### PR TITLE
Add reusable workflow for running `go generate`

### PR DESCRIPTION
**Pull request by webbnh**
## Changes introduced with this PR

This PR provides the common code extracted from the `arcaflow-plugin-sdk-go` and `arcaflow-engine` repositories for running the `go generate` command:  a common, reusable workflow and the wrapper script which it invokes.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/scripts/gogenerate.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+export NOLINT=1
+go generate >/tmp/gogenerate.output 2>/tmp/gogenerate.output
```
Comment (dbutenhof) [inline .github/scripts/gogenerate.sh]: That's actually an interesting bug I have rarely encountered. `stdout` and `stderr` are being *independently* redirected to create the same output file. This doesn't actually work:

1. with `noclobber` on, this fails with "cannot overwrite existing file";
2. with `noclobber` off, this "works" but loses the `stdout` output and captures only `stderr` output

The proper way to redirect both is either `>/tmp/gogenerate.output 2>&1` or just `&>/tmp/gogenerate.output` (which redirects both).

**Review by dbutenhof — CHANGES_REQUESTED** [review]

File: .github/scripts/gogenerate.sh (hunk 1 above)
Comment (webbnh) [reply .github/scripts/gogenerate.sh]: You have an excellent point.  I didn't note this problem when I was moving this code.

> 2\. with `noclobber` off, this "works" but loses the `stdout` output and captures only `stderr` output

In my experiment, it's actually `stderr` which is lost...so, perhaps it's even weirder than we know!

I'll make the change.

Comment (webbnh) [reply .github/scripts/gogenerate.sh]: > it's even weirder than we know!

If you write _once_ to each of `stderr` and `stdout`, then whichever is written second persists, and the other is lost; however, if you alternate writes, then it's even weirder!  Writing to `stderr`, then `stdout`, then `stderr` loses only the first write; likewise writing to `stdout`, then `stderr`, then `stdout`, again loses only the first write.

🥴

Comment (dbutenhof) [reply .github/scripts/gogenerate.sh]: > If you write _once_ to each of `stderr` and `stdout`, then whichever is written second persists, and the other is lost; however, if you alternate writes, then it's even weirder!

Ah: I briefly wondered how that would fall out but decided not to experiment further. That makes sense, in a weird wonderland sort of way ...

**Review by dbutenhof — APPROVED** [review]

**Review by jaredoconnell — APPROVED** [review]

**Pull request merged**